scanner/e2etests/image_test.go

```diff
@@ -0,0 +1,244 @@
+	CheckProvidedExecutables bool `json:"check_provided_executables"`
+}
+
+// TestImage is the E2E test for container image scans.
+func TestImage(t *testing.T) {
+	ctx := context.Background()
+	c, err := client.NewGRPCScannerClient(ctx,
+		client.WithAddress(":8443"),
+		client.WithoutTLSVerify)
+	require.NoError(t, err)
+
+	testCases, err := loadTestCases("testdata/image_tests.json")
+	require.NoError(t, err)
+
+	for _, tc := range testCases {
+		t.Run(fmt.Sprintf("%s/%s", tc.Namespace, tc.Image), func(t *testing.T) {
+			ref, err := name.ParseReference(tc.Image)
+			require.NoError(t, err)
+
+			auth := tc.authConfig()
+			d, err := indexer.GetDigestFromReference(ref, auth)
+			require.NoError(t, err)
+
+			vr, err := c.IndexAndScanImage(ctx, d, auth)
+			require.NoError(t, err)
+
+			expected := &tc.TestWant
+			converted := tc.convertReport(vr)
+			defer func() {
+				if t.Failed() && vr != nil {
+					tc.logReport(t, vr)
+				}
+			}()
+			assert.Equal(t, expected, converted,
+				"The vulnerability report converted did not contain the expected values.")
+		})
+	}
+}
+
+func loadTestCases(s string) ([]*TestCase, error) {
+	f, err := os.Open(s)
+	if err != nil {
+		return nil, err
+	}
+	defer utils.IgnoreError(f.Close)
+	decoder := json.NewDecoder(f)
+	var cases []*TestCase
+	err = decoder.Decode(&cases)
+	if err != nil {
+		return nil, err
+	}
+	return cases, nil
+}
+
+// namespace creates a test case namespace from a vulnerability report.
+func namespace(vr *v4.VulnerabilityReport) string {
+	if len(vr.GetContents().GetDistributions()) == 1 {
+		d := vr.GetContents().GetDistributions()[0]
+		return fmt.Sprintf("%s:%s", d.GetDid(), d.GetVersionId())
+	}
+	return "unknown"
+}
+
+func (tc *TestCase) authConfig() authn.Authenticator {
+	return authn.FromConfig(authn.AuthConfig{
+		Username: os.Getenv(tc.Username),
+		Password: os.Getenv(tc.Password),
+	})
+}
+
+func (tc *TestCase) convertReport(vr *v4.VulnerabilityReport) *TestWant {
+	return &TestWant{
+		Namespace: namespace(vr),
+		Source:    tc.Source,
+		Features:  tc.convertFeatures(vr),
+	}
+}
+
+func (tc *TestCase) convertFeatures(vr *v4.VulnerabilityReport) []Feature {
+	nv := func(n, v string) string {
+		return fmt.Sprintf("%s-%s", n, v)
+	}
+	// Populate map with all features in the test case.
+	feats := make(map[string]*Feature, len(tc.Features))
+	for _, f := range tc.Features {
+		feats[nv(f.Name, f.Version)] = &f
+	}
+	// Convert every expected package in the report.
+	ret := make([]Feature, 0, len(tc.Features))
+	for _, p := range vr.GetContents().GetPackages() {
+		f, ok := feats[nv(p.Name, p.Version)]
+		if !ok {
+			// We only care for features in the test case.
+			continue
+		}
+		ret = append(ret, Feature{
+			Name:            p.GetName(),
+			NamespaceName:   namespace(vr),
+			Version:         p.GetVersion(),
+			Vulnerabilities: tc.convertVulns(vr, p, f),
+			// TODO Pending fields not currently available in the vulnerability report.
```

Comment:
do we have a ticket for this? What is missing that we may want to add?

Reply:
Q1: We do, port E2E tests. Q2: The additional checks the Scanner V2 E2E performs on the fields.